### ISSUE TITLE
Greyed out (inactive) html chunk output

### ISSUE THREAD

**Issue by gregleleu** [rstudio/rstudio#10836]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.02.1 Build 461
    OS Version      :  Monterrey 12.2.1
    R Version       : 4.1.2

### Steps to reproduce the problem
Any chunk that produces HTML based output (examples below with compareDF and leaflet)

````
```{r cars}
library(compareDF)
data(mtcars)

mtcars_minus_10 <- mtcars[1:21,]

a <- compare_df(mtcars_minus_10, mtcars)
create_output_table(a)
```

```{r}
library(leaflet)
leaflet() %>%
  addTiles() %>%  # Add default OpenStreetMap map tiles
  addMarkers(lng=174.768, lat=-36.852, popup="The birthplace of R")
```
````

### Describe the problem in detail
The output is greyed out, "inactive"

CompareDF stays like this, so I can't scroll down when there are many lines
<img width="554" alt="image" src="https://user-images.githubusercontent.com/33321678/159836698-2b66376b-2556-460f-b455-9a3da610d535.png">

But leaflet "reactivates" on hover
<img width="897" alt="image" src="https://user-images.githubusercontent.com/33321678/159836907-ef473ee4-b794-4d14-b1b1-bb0f08d060a4.png">
<img width="857" alt="image" src="https://user-images.githubusercontent.com/33321678/159836935-ef5f3768-c011-4dfc-962e-ce11e8c592b0.png">

### Describe the behavior you expected
Before this the chunk outputs were always "active", and I could scroll down compareDF tables. 
I don't understand why leaflet "reactivates". If this is a feature, it's a quite distracting, maybe wait longer before deactivating?

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@gregleleu Thank you for raising this! What's the most recent version of RStudio Desktop in which this was working for you? Thanks!

**Comment by gregleleu** [comment]
It has been like that for a while unfortunately, I can't tell you precisely :(
It might also have happened when I switched to an M1 mac

**Comment by kevinushey** [comment]
Might be related to https://github.com/rstudio/rstudio/commit/7aac53988df0c7ccd51cce7f40d1718d1da16809#diff-2aecfa93403f791a7a36880f126a6eef20b9392352bf8be504a67a84cc276d15R41.

You can also reproduce with:

```{r}
library(leaflet)

# force some console output
# comment out this line and things seem to work again
print(1)

# then print leaflet widget
leaflet() %>%
  addTiles() %>%  # Add default OpenStreetMap map tiles
  addMarkers(lng=174.768, lat=-36.852, popup="The birthplace of R")
```

If this is done, it's not possible to give focus to the HTML widget.

@GCRev, do you have any ideas?

**Comment by kevinushey** [comment]
This might be worth including as part of the next patch release, since if I understand correctly this bug implies that users will be unable to interact with HTML widgets if they're produced in a chunk that also produces other outputs (e.g. console output), which will be fairly common.

**Comment by gregleleu** [comment]
As this is intended as a feature, would it make sense to make it optional? Or at least to deactivate it when a chunk is popped out (when scrolling the Rmd is not an issue)? 
The experience with having to "reactivate" Leaflet will take some time getting used to, but indeed I used to scroll only on the side to make sure not to hit chunks...

**Comment by kevinushey** [comment]
The underlying issue here might be that HTML widgets are not wrapped in an iframe when presented as part of the "gallery" of chunk output; that is, when there is more than one type of output in a chunk. That might be the root cause here.

**Comment by GCRev** [comment]
@kevinushey this is related to the HTML output scrolling fix, but I can't figure out why it isn't functional in this specific case. The main container of the iframe should have a number of event listeners on it that will search for iframes in its children (anywhere at all in the node tree below it) and "enable" them. In this case, it's not adding those events. I haven't figured out why it isn't adding them yet.

**Comment by GCRev** [comment]
@kevinushey It seems like it has to do with the pagination of multiple outputs. It sets up the events at first -- in this case to the console output window, which is a noop -- and then paging between the two removes all events. I think I have to move the event handlers up a level in the DOM, so they persist when switching pages, or down so that they are added each time the html output is displayed.

**Comment by GCRev** [comment]
Fixed by PR here: https://github.com/rstudio/rstudio/pull/10852

**Comment by mikebessuille** [comment]
@GCRev can you please let Ron know when this has been backported to PT.2 patch?

**Comment by ronblum** [comment]
Verified fixed on in RStudio Desktop 2022.06.0-daily-272 (Spotted Wakerobin) on MacOS 12.3 using all of the examples above as well as from the QA Notes in #10852.

TBD: Verify in Prairie Trillium when ready.

**Comment by ronblum** [comment]
Verified fixed in
- RStudio Desktop 2021.09.3-391 on MacOS 12.3 (Ghost Orchid)
- RStudio Desktop 2022.02.2-481 on MacOS 12.3 (Prairie Trillium)
- RStudio Server 2021.09.3+390 on Ubuntu 20.04 via Chrome and Safari on MacOS 12.3 (Ghost Orchid)
- RStudio Server 2022.02.2-481 on Ubuntu 20.04 via Chrome and Safari on MacOS 12.3 (Ghost Orchid)

Confirmed that
- when hovered over and clicked, the output is active
- when not hovered over, or clicking in the code and using up and down arrows, the output is skipped
